httputil/client.go

```diff
@@ -141,10 +141,13 @@ func (d *dialTLS) addLocalSSLCertificates() (err error) {
 }
 
 type ClientOptions struct {
-	Timeout       time.Duration
-	TLSConfig     *tls.Config
-	MayLogBody    bool
-	Proxy         func(*http.Request) (*url.URL, error)
+	Timeout    time.Duration
+	TLSConfig  *tls.Config
+	MayLogBody bool
+
+	Proxy              func(*http.Request) (*url.URL, error)
+	ProxyConnectHeader http.Header
```

Comment:
Remark about the new feature (I think this applies across the whole pull request).

Reply:
it's not a new feature, I had to move the responsibility for this because httputil shouldn't use snapdenv:

https://github.com/snapcore/snapd/pull/8264/files#diff-f03e0337b0718cee67b29109c56280e8R165

Reply:
Ah, I see now. Thank you